Module#03/api/src/main/java/com/epam/esm/controller/CertificateController.java

```diff
@@ -1,78 +1,90 @@
 package com.epam.esm.controller;
 
+import com.epam.esm.assembler.CertificateAssembler;
 import com.epam.esm.criteria.Criteria;
-import com.epam.esm.criteria.SortOrder;
-import com.epam.esm.dto.CertificateWithoutTagDto;
 import com.epam.esm.dto.CertificateDto;
+import com.epam.esm.dto.Dto;
+import com.epam.esm.dto.TagDto;
 import com.epam.esm.service.CertificateService;
 import lombok.RequiredArgsConstructor;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.PatchMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
 
-import java.time.Instant;
 import java.util.List;
 
 import static org.springframework.http.HttpStatus.CREATED;
-import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
 import static org.springframework.http.MediaType.APPLICATION_JSON_VALUE;
 
 @RestController
 @RequiredArgsConstructor
-@RequestMapping(value = "/certificates")
+@RequestMapping(value = "/api/certificates")
```

Comment:
'/api' - path can be applied the hole project with 'server.servlet.context-path=/api' configuration in application.properties file. Please take a look on it

Reply:
config servlet.context path was added